### PR TITLE
AuthN: Add id token to gRPC client interceptors

### PR DESCRIPTION
**Pull request by gamab**
This PR allows users to add id tokens to the grpc metadata in the outgoing context.

Additionally, to cover future usage, this PR allows adding additional decorators to the metadata.

### CODE REVIEW

File: authn/grpc_client_interceptors.go [hunk 1]
```diff
@@ -8,9 +8,11 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-var DefaultAccessTokenMetadataKey = "X-Access-Token"
+var (
```
Comment (linoman) [inline authn/grpc_client_interceptors.go:11]: This could perfectly be `const`

File: authn/grpc_client_interceptors.go [hunk 2]
```diff
@@ -32,8 +37,11 @@ type GrpcClientConfig struct {
 type GrpcClientInterceptor struct {
 	cfg         *GrpcClientConfig
 	tokenClient TokenExchanger
+	mdFns       []MetadataDecorator
```
Comment (linoman) [inline authn/grpc_client_interceptors.go:40]: Nit: I'm not a fan of abbreviations; it's hard to understand this, even with context.

File: authn/grpc_client_interceptors.go [hunk 3]
```diff
@@ -100,5 +130,12 @@ func (gci *GrpcClientInterceptor) wrapContext(ctx context.Context) (context.Cont
 
 	md.Set(gci.cfg.AccessTokenMetadataKey, token.Token)
 
+	for _, fn := range gci.mdFns {
+		md, err = fn(md)
```
Comment (linoman) [inline authn/grpc_client_interceptors.go:134]: I am confused by this code. Why is `md` being assigned to `md` again?

**Review by linoman — APPROVED** [review]
Overall, it looks good. I left some nits and questions. It would be great if we could address them before moving on.

**Review by Jguer — APPROVED** [review]

**Review by Jguer — APPROVED** [review]

File: authn/grpc_client_interceptors.go (hunk 2 above)
Comment (gamab) [reply authn/grpc_client_interceptors.go]: Accounted for.

**Pull request merged**